sklearn/tree/_criterion.pyx

```diff
@@ -230,10 +261,19 @@ cdef class ClassificationCriterion(Criterion):
         self.start = 0
         self.pos = 0
         self.end = 0
+        self.missing_direction = 2  # Ignore the missing values by default
 
         self.n_outputs = n_outputs
+
+        self.n_missing = 0
+        self.weighted_n_missing = 0
+
+        self.n_nonmissing = 0
+        self.weighted_n_nonmissing = 0
```

Comment:
The right and left partition's weights are updated at each iteration and these values are stored in separate variables, which are used for the impurity computation. So at the level of impurity computation, the existence of these examples are abstract.